timelock-impl/src/main/java/com/palantir/atlasdb/timelock/clock/ClockSkewMonitor.java

```diff
@@ -80,31 +80,49 @@ public void runInBackground() {
     }
 
     private void runOnce() {
-        try {
-            runInternal();
-        } catch (Throwable t) {
-            events.exception(t);
-        }
+        Map<String, RequestTime> newRequests = getRemoteRequestTimes();
+        checkAndUpdatePreviousRequestTimes(newRequests);
     }
 
-    private void runInternal() {
-        clocksByServer.forEach((server, remoteClockService) -> {
-            long localTimeAtStart = localClockService.getSystemTimeInNanos();
-            long remoteSystemTime = remoteClockService.getSystemTimeInNanos();
-            long localTimeAtEnd = localClockService.getSystemTimeInNanos();
+    private Map<String, RequestTime> getRemoteRequestTimes() {
+        Map<String, RequestTime> newRequestTimes = Maps.newHashMap();
 
-            RequestTime previousRequest = previousRequestsByServer.get(server);
-            RequestTime newRequest = RequestTime.builder()
-                    .localTimeAtStart(localTimeAtStart)
-                    .localTimeAtEnd(localTimeAtEnd)
-                    .remoteSystemTime(remoteSystemTime)
-                    .build();
+        clocksByServer.forEach((host, clockService) -> {
+            try {
+                RequestTime requestTime = getNewRequestTime(clockService);
+                newRequestTimes.put(host, requestTime);
+            } catch (Throwable t) {
+                events.exception(t);
+            }
+        });
+        return newRequestTimes;
+    }
+
+    private RequestTime getNewRequestTime(ReversalDetectingClockService remoteClockService) {
+        long localTimeAtStart = localClockService.getSystemTimeInNanos();
+        long remoteSystemTime = remoteClockService.getSystemTimeInNanos();
+        long localTimeAtEnd = localClockService.getSystemTimeInNanos();
 
+        return RequestTime.builder()
+                .localTimeAtStart(localTimeAtStart)
+                .localTimeAtEnd(localTimeAtEnd)
+                .remoteSystemTime(remoteSystemTime)
+                .build();
+    }
+
+    private void checkAndUpdatePreviousRequestTimes(Map<String, RequestTime> newRequests) {
+        newRequests.forEach((remoteHost, newRequest) -> {
+            RequestTime previousRequest = previousRequestsByServer.get(remoteHost);
             if (previousRequest != null) {
-                new ClockSkewComparer(server, events, previousRequest, newRequest).compare();
+                try {
+                    new ClockSkewComparer(remoteHost, events, previousRequest, newRequest).compare();
+                } catch (IllegalArgumentException exception) {
```

Comment:
cool, this also fixes the bug where we don't update `previousRequestByServer` if we get an exception here